hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/datanodes/datanodes.tsx

```diff
@@ -339,7 +341,14 @@ export class Datanodes extends React.Component<Record<string, object>, IDatanode
       loading: true,
       selectedColumns: this._getSelectedColumns(prevState.selectedColumns)
     }));
-    axios.get('/api/v1/datanodes').then(response => {
+
+    if (typeof cancelToken != typeof undefined){
+      //Some existing request might be there so we will cancel that before sending new request
+      cancelToken.cancel("Cancelling old datanode request");
```

Comment:
If "componentWillUnmount" will cancel the request on changing the view, then what is the possibility of older request still be there ?

Reply:
This is handling cases, when say autoReload happens, it will call _loadData(), but the component is not unmounted.

Reply:
> This is handling cases, when say autoReload happens, it will call _loadData(), but the component is not unmounted.

So is it possible to override _loadData() and call "componentWillUnmount" explicitly inside it ?

Reply:
We cannot explicitly call componentWillUnmount. It is auto triggered before the component is just about to be removed from DOM. Take it as the equivalent of say stop() for a thread, only we cannot explicitly do this.componentWillUnmount() to call it. So I had it handled in _loadData().

Reply:
As discussed, pls refactor it to reduce duplicity of code.